### DISCUSSION TITLE
MongoDB index not created with `wildcardProjection`

### DISCUSSION THREAD

**Discussion by pooledge** [payloadcms/payload#4417] [Q&A]
### Describe the Bug

Programmatically creating a MongoDB text index with a wildcard works only if `wildcardProjection` key is not present within `options`.

Background: I'd like to [text index](https://github.com/payloadcms/payload/discussions/4079) Lexical text content. Since I cannot know how deep it goes my plan was to wildcard it, though I'd probably go with explicitly indexing text nodes a couple of levels deep instead:
```ts
blocks.content.root.children.children.text
blocks.content.root.children.children.children.text
blocks.content.root.children.children.children.children.text
...
```

### To Reproduce

After `payload.init()` in `server.ts`:
```ts
payload.db.collections['my_collection_name'].schema.index(
	{
		'$**': 'text',
	},
	{
		wildcardProjection: {
			'meta.title': 1
		},
	}
);
```
won't create the desired index. Commenting out the `wildcardProjection` entirely creates an index as expected.

### Payload Version

2.3.1

### Adapters and Plugins

@payloadcms/db-mongodb v1.1.0

**Comment by DanRibbens** [comment]
I moved this to a bug since Payload has some helpers for creating indexes, but othis is out scope for what we give.

I'm not too familiar with the widlcardProjection option in MongoDB. It looks like this is how you need to specify indexes in older versions of MongoDB (4.2). What version of MongoDB are you using as it seems the formats for wildcard indexes has changed.

Mongoose might have some more info, but I'm reading the MongoDB documents for the current version and it doesn't look like this is how you do it anymore https://www.mongodb.com/docs/v7.0/core/indexes/index-types/index-wildcard/.

**Reply by pooledge** [reply]
Hmm it seems [it's still here](https://www.mongodb.com/docs/v7.0/core/indexes/index-types/index-wildcard/create-wildcard-index-multiple-fields/#std-label-create-wildcard-index-multiple-fields) or do I miss smth?

**Reply by pooledge** [reply]
I don't think it should be prioritised though. It's an edge case you can basically mitigate with setting an approximate depth explicitly:
```ts
export const createLexicalTextIndex = (node: string, depth: number = 5) => {
	let obj = {};

	for (let i = 1; i <= depth; i++) {
		const key = [`${node}.root`, Array(i).fill('children'), 'text']
			.flat()
			.join('.');

		obj[key] = 'text';
	}

	return obj;
};
```
